### ISSUE TITLE
Error while removing species

### ISSUE THREAD

**Issue by MrTomRod** [davidemms/OrthoFinder#311]
I ran OrthoFinder 2.3.7 for 67 fasta-files. Then, I wanted to remove all but the first 10 files. First, I commented out the unwanted files in `/input/OrthoFinder/Results_Nov22_1/WorkingDirectory/SpeciesIDs.txt`.

Then I ran this:
```
# orthofinder -b /input/OrthoFinder/Results_Nov22_1/WorkingDirectory/ -t 8 -a 4 -S diamond

OrthoFinder version 2.3.7 Copyright (C) 2014 David Emms

2019-11-22 12:45:26 : Starting OrthoFinder
8 thread(s) for highly parallel tasks (BLAST searches etc.)
4 thread(s) for OrthoFinder algorithm

Checking required programs are installed
----------------------------------------
Test can run "mcl -h" - ok
Test can run "fastme -i /input/OrthoFinder/Results_Nov22_1/WorkingDirectory/SimpleTest.phy -o /input/OrthoFinder/Results_Nov22_1/WorkingDirectory/SimpleTest.tre" - ok
Using previously calculated BLAST results in /input/OrthoFinder/Results_Nov22_1/WorkingDirectory/

Running OrthoFinder algorithm
-----------------------------
2019-11-22 12:45:27 : Initial processing of each species
2019-11-22 12:45:28 : Initial processing of species 2 complete
2019-11-22 12:45:28 : Initial processing of species 3 complete
2019-11-22 12:45:28 : Initial processing of species 0 complete
2019-11-22 12:45:29 : Initial processing of species 1 complete
2019-11-22 12:45:30 : Initial processing of species 4 complete
2019-11-22 12:45:30 : Initial processing of species 6 complete
2019-11-22 12:45:31 : Initial processing of species 5 complete
2019-11-22 12:45:31 : Initial processing of species 7 complete
2019-11-22 12:45:32 : Initial processing of species 8 complete
2019-11-22 12:45:33 : Initial processing of species 9 complete
2019-11-22 12:45:36 : Connected putative homologues
2019-11-22 12:45:36 : Written final scores for species 2 to graph file
2019-11-22 12:45:37 : Written final scores for species 3 to graph file
2019-11-22 12:45:37 : Written final scores for species 0 to graph file
2019-11-22 12:45:37 : Written final scores for species 1 to graph file
2019-11-22 12:45:37 : Written final scores for species 4 to graph file
2019-11-22 12:45:37 : Written final scores for species 6 to graph file
2019-11-22 12:45:37 : Written final scores for species 7 to graph file
2019-11-22 12:45:37 : Written final scores for species 5 to graph file
2019-11-22 12:45:38 : Written final scores for species 8 to graph file
2019-11-22 12:45:38 : Written final scores for species 9 to graph file
2019-11-22 12:45:40 : Ran MCL

Writing orthogroups to file
---------------------------
OrthoFinder assigned 25065 genes (92.9% of total) to 4002
orthogroups. Fifty percent of all genes were in orthogroups
with 8 or more genes (G50 was 8) and were contained in the
largest 1227 orthogroups (O50 was 1227). There were 762
orthogroups with all species present and 601 of these
consisted entirely of single-copy genes.

2019-11-22 12:45:41 : Done orthogroups

Analysing Orthogroups
=====================

Calculating gene distances
--------------------------
2019-11-22 12:45:52 : Done

Inferring gene and species trees
--------------------------------
2019-11-22 12:45:53 : Done 0 of 2675
2019-11-22 12:45:53 : Done 1000 of 2675
2019-11-22 12:45:54 : Done 2000 of 2675

762 trees had all species present and will be used by STAG to infer the species tree

Best outgroup(s) for species tree
---------------------------------
2019-11-22 12:46:03 : Starting STRIDE
2019-11-22 12:46:04 : Done STRIDE
Observed 35 well-supported, non-terminal duplications. 35 support the best root and 0 contradict it.
Best outgroup for species tree:
  STRAIN1, STRAIN2, STRAIN3

Reconciling gene trees and species tree
---------------------------------------
Outgroup: STRAIN1, STRAIN2, STRAIN3
2019-11-22 12:46:04 : Starting Recon and orthologues
2019-11-22 12:46:04 : Starting OF Orthologues
Traceback (most recent call last):
  File "orthofinder/orthofinder.py", line 1663, in <module>
  File "orthofinder/orthofinder.py", line 1482, in GetOrthologues
  File "orthofinder/scripts/orthologues.py", line 1002, in OrthologuesWorkflow
  File "orthofinder/scripts/orthologues.py", line 800, in ReconciliationAndOrthologues
  File "orthofinder/scripts/trees2ologs_of.py", line 638, in DoOrthologuesForOrthoFinder
  File "orthofinder/scripts/orthologues.py", line 137, in Spec_SeqDict
  File "orthofinder/scripts/orthologues.py", line 137, in <dictcomp>
KeyError: '64'
[1458] Failed to execute script orthofinder
```

Can you help me?

**Comment by davidemms** [comment]
Hi

I've realised that this can be caused if you comment out with an extra character or space, e.g. the commenting out of species 3 is incorrect here:
```
0: Mycoplasma_agalactiae.faa
1: Mycoplasma_gallisepticum.faa
#2: Mycoplasma_genitalium.faa
# 3: Mycoplasma_hyopneumoniae.faa
```

Could something like that be the cause? If this isn't the problem then let me know. I will also submit a fix for this issue.

All the best
David

**Comment by TheBigFatTony** [comment]
Exactly, I had leading spaces, like here: `# 3: Mycoplasma_hyopneumoniae.faa`

It's fixed in version 2.3.8. Btw, I updated the Dockerfile for this version: https://github.com/MrTomRod/Orthofinder-Dockerfile

**Comment by MrTomRod** [comment]
Exactly, I had leading spaces, like here: `# 3: Mycoplasma_hyopneumoniae.faa`

It's fixed in version 2.3.8, thanks! Btw, I updated the Dockerfile for this version: https://github.com/MrTomRod/Orthofinder-Dockerfile
